core/src/types/transaction/wrapper.rs

```diff
@@ -171,6 +173,8 @@ pub mod wrapper_tx {
         pub epoch: Epoch,
         /// Max amount of gas that can be used when executing the inner tx
         pub gas_limit: GasLimit,
+        /// The optional, unencrypted, unshielding tx for fee payment
+        pub unshield: Option<Tx>,
```

Comment:
I think rather than using the `Tx` type here which allows arbitrary data and code, which would require some validation in the protocol (and also bloats the payload), it would be better to only attach masp `Transaction` here.

Reply:
Agreed

Reply:
Yes, this should be easy to implement given that we've already added the support for tx hashes in #1297. We can load the wasm code before running the operation. Still, some validation will be necessary even though we can get rid of the check on the content of the unshielding tx that will be done now with the type system